packages/material-ui/src/Slider/SliderBase.js

```diff
@@ -0,0 +1,904 @@
+        style={trackStyle}
+      />
+      <input value={values.join(',')} name={name} type="hidden" />
+      {marks.map((mark, index) => {
+        const percent = valueToPercent(mark.value, min, max);
+        const style = axisProps[axis].offset(percent);
+
+        let markActive;
+        if (track === false) {
+          markActive = values.indexOf(mark.value) !== -1;
+        } else {
+          markActive =
+            (track === 'normal' &&
+              (range
+                ? mark.value >= values[0] && mark.value <= values[values.length - 1]
+                : mark.value <= values[0])) ||
+            (track === 'inverted' &&
+              (range
+                ? mark.value <= values[0] || mark.value >= values[values.length - 1]
+                : mark.value >= values[0]));
+        }
+
+        return (
+          <React.Fragment key={mark.value}>
+            <Mark
+              style={style}
+              data-index={index}
+              {...stateAndProps}
+              {...markProps}
+              className={clsx(utilityClasses.mark, markProps.className, {
+                [getUtilityClass('markActive')]: markActive,
+              })}
+              markActive={markActive}
+            />
+            {mark.label != null ? (
+              <MarkLabel
+                aria-hidden
+                data-index={index}
+                style={style}
+                {...stateAndProps}
+                {...markLabelProps}
+                className={clsx(utilityClasses.markLabel, markLabelProps.className, {
+                  [getUtilityClass('markLabelActive')]: markActive,
+                })}
+                markLabelActive={markActive}
+              >
+                {mark.label}
+              </MarkLabel>
+            ) : null}
+          </React.Fragment>
+        );
+      })}
+      {values.map((value, index) => {
+        const percent = valueToPercent(value, min, max);
+        const style = axisProps[axis].offset(percent);
+
+        return (
+          <ValueLabel
+            key={index}
+            valueLabelFormat={valueLabelFormat}
+            valueLabelDisplay={valueLabelDisplay}
+            value={
+              typeof valueLabelFormat === 'function'
+                ? valueLabelFormat(scale(value), index)
+                : valueLabelFormat
+            }
+            index={index}
+            open={open === index || active === index || valueLabelDisplay === 'on'}
+            disabled={disabled}
+            {...stateAndProps}
+            {...valueLabelProps}
+            className={clsx(utilityClasses.valueLabel, valueLabelProps.className)}
+          >
+            <Thumb
+              {...stateAndProps}
+              {...thumbProps}
+              className={clsx(utilityClasses.thumb, thumbProps.className, {
+                ['Mui-active']: active === index,
+                [getUtilityClass('focusVisible')]: focusVisible === index,
```

Comment:
Considering it's meant to replace `:focus-visible`:

```suggestion
                ['Mui-focusVisible']: focusVisible === index,
```

We also list it in https://material-ui.com/customization/components/#pseudo-classes.